src/mlpack/methods/ann/loss_functions/l1_loss_impl.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file l1_loss_impl.hpp
+ * @author Himanshu Pathak
+ *
+ * Implementation of the L1 Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_L1_LOSS_IMPL_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_L1_LOSS_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "l1_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+L1Loss<InputDataType, OutputDataType>::L1Loss(
+  const bool mean):
+  mean(mean)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+double L1Loss<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, const TargetType&& target)
+{ 
+  if (mean)
+  {
+    return arma::sum(arma::abs(input - target)) / input.n_elem;
+  }
+  return arma::sum(arma::abs(input - target));
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType, typename OutputType>
+void L1Loss<InputDataType, OutputDataType>::Backward(
+    const InputType&& input,
+    const TargetType&& target,
+    OutputType&& output)
+{
+  for (size_t i = 0; i < input.n_elem; ++i)
+    output[i] = (input[i] - target[i])/arma::abs(input[i] - target[i]);
+}
```

Comment:
I think sign of ```input[i]-target[i]``` is what we need here. We can then use ```arma::sign()``` and I believe it will be faster. What do you say?

Reply:
@mrityunjay-tripathi Thanks for suggestion I was also thinking of something like this but not able to find in arma docs